pages/api/v1/preimages/latest.ts

```diff
@@ -0,0 +1,89 @@
+// Copyright 2019-2025 @polkassembly/polkassembly authors & contributors
+// This software may be modified and distributed under the terms
+// of the Apache-2.0 license. See the LICENSE file for details.
+import { NextApiHandler } from 'next';
+
+import withErrorHandling from '~src/api-middlewares/withErrorHandling';
+import { isValidNetwork } from '~src/api-utils';
+import { MessageType } from '~src/auth/types';
+import { GET_LATEST_PREIMAGES } from '~src/queries';
+import { IApiResponse } from '~src/types';
+import apiErrorWithStatusCode from '~src/util/apiErrorWithStatusCode';
+import fetchSubsquid from '~src/util/fetchSubsquid';
+import messages from '~src/util/messages';
+
+export interface IPreimageData {
+    hash: string;
+    deposit: string;
+    createdAtBlock: number;
+    length: number;
+    method: string;
+    proposedCall: any;
+    proposer: string;
+    section: string;
+    status: string;
+    message?: string;
+}
+
+interface IPrams{
+  hash: string;
+  network: string;
+}
+export async function getLatestPreimage(params:IPrams ): Promise<IApiResponse<IPreimageData | MessageType>> {
+	try {
+		const { hash , network } = params;
+		if(!network || !isValidNetwork(network)) {
+			throw apiErrorWithStatusCode('Invalid network in request header', 400);
+		}
+		if(!hash) {
+			throw apiErrorWithStatusCode('Invalid hash', 400);
+		}
+		const subsquidRes = await fetchSubsquid({
+			network,
+			query: GET_LATEST_PREIMAGES,
+			variables: { hash_eq: hash }
+		});
+
+		if (subsquidRes && subsquidRes.data && subsquidRes.data.preimages) {
+			const preimages = subsquidRes.data.preimages;
+			if (preimages && Array.isArray(preimages) && preimages[0]) {
+				const preimage: IPreimageData = preimages[0];
+				return {
+					data: JSON.parse(JSON.stringify(preimage)),
```

Comment:
prefer using `data: structuredClone(preimage);` syntax instead of `data: JSON.parse(JSON.stringify(preimage))`

Reply:
moreover is the `data: JSON.parse(JSON.stringify(preimage))` necessary ?